### PR TITLE
Chapter 8: Basic login

### PR DESCRIPTION
**Pull request by thaotonto**
screen shot 2018-07-23 at 1 57 39 pm

### CODE REVIEW

File: app/views/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<% provide :title, t("sessions.new.login") %>
+<h1><%= t "sessions.new.login" %></h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for :session, url: login_path do |f| %>
+
+      <%= f.label :email, t(".email") %>
+      <%= f.email_field :email, class: 'form-control' %>
```
Comment (ducbet) [inline app/views/sessions/new.html.erb]: "

File: app/views/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+      <%= f.label :password, t(".password") %>
+      <%= f.password_field :password, class: 'form-control' %>
```
Comment (ducbet) [inline app/views/sessions/new.html.erb]: "

File: app/helpers/sessions_helper.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (DuongNT1512) [inline app/helpers/sessions_helper.rb]: `current_user.present?`

File: app/models/user.rb [hunk 4]
```diff
@@ -11,6 +11,15 @@ class User < ApplicationRecord
     length: {minimum: Settings.user_setting.password_minimum_length}
   has_secure_password
 
+  def self.digest string
```
Comment (DuongNT1512) [inline app/models/user.rb]: dunfg dạng class << self nhé

File: app/models/user.rb [hunk 5]
```diff
@@ -11,6 +11,15 @@ class User < ApplicationRecord
     length: {minimum: Settings.user_setting.password_minimum_length}
   has_secure_password
 
+  def User.digest string
```
Comment (DuongNT1512) [inline app/models/user.rb]: class << self

**Pull request merged**